src/v/cluster/rm_stm.cc

```diff
@@ -956,14 +956,25 @@ ss::future<tx_errc> rm_stm::do_abort_tx(
           "Error \"{}\" on replicating pid:{} abort batch",
           r.error(),
           pid);
-        co_return tx_errc::unknown_server_error;
+        if (_c->is_leader() && _c->term() == synced_term) {
+            co_await _c->step_down("abort_tx replication error");
+        }
+        co_return tx_errc::timeout;
     }
     if (_mem_state.last_end_tx < r.value().last_offset) {
         _mem_state.last_end_tx = r.value().last_offset;
     }
 
     if (!co_await wait_no_throw(r.value().last_offset, timeout)) {
-        co_return tx_errc::unknown_server_error;
+        vlog(
+          _ctx_log.trace,
+          "Can't wait until abort batch pid:{} tx_seq:{} is applied",
```

Comment:
nit: this log statement is a little bit confusing, should it be saying `error waiting for` ? or is it the case that in current state we can not wait for abort_tx to be applied ?

Reply:
fixed